lib/puppet/pops/loader/task_instantiator.rb

```diff
@@ -84,7 +37,7 @@ def self.create(loader, typed_name, source_refs)
         arguments[key] = value unless arguments.key?(key)
       end
 
-      Types::TypeFactory.task.from_hash(arguments)
```

Comment:
@hlindberg how strict should we be about making changes to interfaces with `Puppet[:task]` enabled?

Reply:
Somewhat - there are  at least one user (R.I) that does advanced things ;-) 
I am more concerned that this seems to make Task just be some kind of Ruby runtime object rather than something described by Pcore. Do we ever have puppet code that references a Task instance?

Is this all because a Pcore object errors if it gets extra parameters? If so, that seems like something that can be fixed.

Reply:
I'm not aware of puppet code ever referencing a Task instance. But if we can allow Pcore objects to ignore extra parameters, we can keep it. I'll want to start using it to validate tasks we load via the InfoService as well.

Reply:
I'm really not sure how a Task data type in Puppet makes any sense.